python/dpu_utils/codeutils/treesitter/parser.py

```diff
@@ -0,0 +1,66 @@
+import logging
+import os
+from tempfile import TemporaryDirectory
+
+from tree_sitter import Language, Parser
+
+__all__ = ["parser_for"]
+
+_LANGUAGE_REPOS = {
+    "bash": ("https://github.com/tree-sitter/tree-sitter-bash", ""),
+    "c": ("https://github.com/tree-sitter/tree-sitter-c", ""),
+    "c_sharp": ("https://github.com/tree-sitter/tree-sitter-c-sharp", ""),
+    "css": ("https://github.com/tree-sitter/tree-sitter-css", ""),
+    "cpp": ("https://github.com/tree-sitter/tree-sitter-cpp", ""),
+    "html": ("https://github.com/tree-sitter/tree-sitter-html", ""),
+    "go": ("https://github.com/tree-sitter/tree-sitter-go", ""),
+    "java": ("https://github.com/tree-sitter/tree-sitter-java", ""),
+    "javascript": ("https://github.com/tree-sitter/tree-sitter-javascript", ""),
+    "julia": ("https://github.com/tree-sitter/tree-sitter-julia", ""),
+    "php": ("https://github.com/tree-sitter/tree-sitter-php", ""),
+    "python": ("https://github.com/tree-sitter/tree-sitter-python", ""),
+    "ruby": ("https://github.com/tree-sitter/tree-sitter-ruby", ""),
+    "rust": ("https://github.com/tree-sitter/tree-sitter-rust", ""),
+    "scala": ("https://github.com/tree-sitter/tree-sitter-scala", ""),
+    "typescript": ("https://github.com/tree-sitter/tree-sitter-typescript", "typescript"),
+}
+
+LIBRARY_DIR = os.path.join(os.path.dirname(__file__), "build", "treesitter-lib.so")
+TREE_SITTER_LANG_VER = "v0.19.0"

```

Comment:
Do we want to make this a constant here or somehow configurable downstream?

Reply:
Maybe/eventually, but I'll ignore this for now as I am not sure how these grammars/tree-sitter evolves. These seem to be somehow tied to tree-sitter (each version requiring a particular grammar version), but we'll have to iterate through this in the future if/when new tree-sitter versions come up.